work_queue/src/work_queue.c

```diff
@@ -827,6 +855,127 @@ static int get_transfer_wait_time(struct work_queue *q, struct work_queue_worker
 	return timeout;
 }
 
+static int factory_trim_workers(struct work_queue *q, struct work_queue_factory_info *f)
+{
+	// Iterate through all workers and shut idle ones down
+	char *factory_name = f->name;
+	struct work_queue_worker *w;
+	char *key;
+	int trimmed_workers = 0;
+
+	hash_table_firstkey(q->worker_table);
+	while( f->connected_workers > f->max_workers &&
+			hash_table_nextkey(q->worker_table, &key, (void **) &w) ) {
+		if( !strcmp(factory_name, w->factory_name) &&
```

Comment:
`w->factory_name && !strcmp(factory_name, w->factory_name) &&`

strcmp will segfault if w->factory_name is not set otherwise.

Reply:
Solving this by making default factory_name `"unknown"`

Reply:
I don't recommend setting a default factory name. It will get confusing, and we have to remember that "unknown" is a special value. Let's just check whether w->factory_name is different from NULL.